library/Core.cpp

```diff
@@ -1506,6 +1503,12 @@ bool Core::InitMainThread() {
     std::cerr << "DFHack build: " << Version::git_description() << "\n"
          << "Starting with working directory: " << Filesystem::getcwd() << std::endl;
 
+    std::cerr << "Binding to SDL.\n";
+    if (!DFSDL::init(con)) {
+        fatal("cannot bind SDL libraries");
+        return false;
+    }
+
```

Comment:
I'm hoping that moving this block to InitMainThread() is safe. This is needed so that the SDL function is bound before we call fatal() if the DF version is unknown.

I'm realizing now that there is also a fatal() call inside this block, so I should protect it somehow...

Reply:
the only risk i can see is if somehow `InitSimulationThread` executed before, or in parallel with, `InitMainThread`

we call `initMainThread` from `dfhooks_init` and DF calls this before it starts the simulation thread, so it'll be completed before DF's simulation thread can ever call `dfhooks_update` and that appears to be the only code path that can reach `initSimulationThread`

so i think this should be safe, from a thread safety standpoint